Assets/Scripts/Models/Utility.cs

```diff
@@ -0,0 +1,802 @@
+        // be assigned to a tile.  So just read extra data if we have any.
+        if (!reader.IsEmptyElement)
+        {
+            ReadXmlParams(reader);
+        }
+    }
+
+    /// <summary>
+    /// Reads the XML for parameters that this utility has and assign it to the utility.
+    /// </summary>
+    /// <param name="reader">The reader to read the parameters from.</param>
+    public void ReadXmlParams(XmlReader reader)
+    {
+        // X, Y, and objectType have already been set, and we should already
+        // be assigned to a tile.  So just read extra data.
+        utilParameters = Parameter.ReadXml(reader);
+    }
+
+    /// <summary>
+    /// Gets the utility's Parameter structure from a string key.
+    /// </summary>
+    /// <returns>The Parameter value..</returns>
+    public Parameter GetParameters()
+    {
+        return utilParameters;
+    }
+
+    /// <summary>
+    /// How many jobs are linked to this utility.
+    /// </summary>
+    /// <returns>The number of jobs linked to this utility.</returns>
+    public int JobCount()
+    {
+        return jobs.Count;
+    }
+
+    /// <summary>
+    /// Link a job to the current utility.
+    /// </summary>
+    /// <param name="job">The job that you want to link to the utility.</param>
+    public void AddJob(Job job)
+    {
+        // FIXME: Do we even need this?
+//        job.utility = this;
+        jobs.Add(job);
+        job.OnJobStopped += OnJobStopped;
+        World.Current.jobQueue.Enqueue(job);
+    }
+
+    /// <summary>
+    /// Cancel all the jobs linked to this utility.
+    /// </summary>
+    public void CancelJobs()
+    {
+        Job[] jobsArray = jobs.ToArray();
+        foreach (Job job in jobsArray)
+        {
+            job.CancelJob();
+        }
+    }
+
+    /// TODO: Refactor this when the new job system is implemented
+    public void ResumeJobs()
+    {
+        Job[] jobsArray = pausedJobs.ToArray();
+        foreach (Job job in jobsArray)
+        {
+            AddJob(job);
+            pausedJobs.Remove(job);
+        }
+    }
+
+    /// TODO: Refactor this when the new job system is implemented
+    public void PauseJobs()
+    {
+        Job[] jobsArray = jobs.ToArray();
+        foreach (Job job in jobsArray)
+        {
+            pausedJobs.Add(job);
+            job.CancelJob();
+        }
+    }
+
+    /// <summary>
+    /// Deconstructs the utility.
+    /// </summary>
+    public void Deconstruct(Utility utility)
+    {
+        int x = Tile.X;
+        int y = Tile.Y;
+        int fwidth = 1;
+        int fheight = 1;
+        bool linksToNeighbour = false;
+        if (Tile.Utilities != null)
+        {
+            linksToNeighbour = utility.LinksToNeighbour;
+            utility.CancelJobs();
+        }
+
+        // We call lua to decostruct
+        // TODO: EventActions doesn't currently allow anything but Utility Fix this when implemented
+//        EventActions.Trigger("OnUninstall", this);
```

Comment:
Since the FunctionsManager was merged. You can now generalize EventActions as you wanted.